erroranalysis/erroranalysis/_internal/surrogate_error_tree.py

```diff
@@ -302,7 +302,7 @@ def get_surrogate_booster_local(filtered_df, analyzer, is_model_analyzer,
     if is_pandas:
         true_y = true_y.to_numpy()
     else:
-        input_data = input_data.to_numpy()
+        input_data = input_data.to_numpy(copy=True)
```

Comment:
Are there any other places in the repo where data.to_numpy is used and this change needs to be done there too?

Reply:
It is hard to validate all the cases. So making changes and adding tests for the cases reported by customers.